File: unit2/caesar/caesar.c
/*
 * caesar.c by Yesha Thakkar
 * November 7, 2018
 * Encrypts messages using the Caesar Cipher
 */

#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>


void ciphertext_converter(char letter, int ascii_number, int key);

int main(int argc, string argv[])
{
    // Test the number of arguments
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }

    // User input statement
    string plaintext = get_string("plaintext: ");

    // Converts the key to an integer
    int key = atoi(argv[1]);

    // Prints "ciphertext"
    printf("ciphertext: ");

    // Tests whether a character is a letter and decides what operation to perform
    for (int i = 0, n = strlen(plaintext); i < n; i++)
    {
        if (isalpha(plaintext[i]))
        {
            // Uppercase letter test
            if (isupper(plaintext[i]))
            {
                ciphertext_converter(plaintext[i], 'A', key);
            }
            // Lowercase letter test
            else
            {
                ciphertext_converter(plaintext[i], 'a', key);
            }
        }
        // Handles spaces and punctuation
        else
        {
            printf("%c", plaintext[i]);
        }
    }

    // Prints a new line
    printf("\n");

    // Success!!
    return 0;
}

// Function turns individual characters into ciphertext using modulo
void ciphertext_converter(char letter, int ascii_number, int key)
{
    int character_number = letter - ascii_number;
    int ciphertext_number = (character_number + key) % 26;
    char ciphertext_letter = (char)(ciphertext_number + ascii_number);
    printf("%c", ciphertext_letter);
}

File: unit6/sentimental/caesar.py
import cs50
import sys


def main():
    # Ensure that arguments exist and are of the right type
    while True:
        if (len(sys.argv) != 2):
            print("Usage: python caesar.py key")
            sys.exit(1)
        else:
            plaintext = cs50.get_string("Plaintext: ")
            key = int(sys.argv[1])
            break

    # Sends each letter to the converter function
    ciphertext = ""
    for i in plaintext:
        if i.isalpha():
            value = ciphertext_converter(i, key)
            ciphertext += value
        else:
            ciphertext += i
    print("ciphertext: " + ciphertext)


# Converter function first checks case then converts
def ciphertext_converter(character, key):
    if character.isupper():
        return chr(((ord(character) - 65 + key) % 26) + 65)
    else:
        return chr(((ord(character) - 97 + key) % 26) + 97)


if __name__ == "__main__":
    main()